### ISSUE TITLE
You can no longer edit or delete modding discussion posts

### ISSUE THREAD

**Issue by nl-tatatat** [ppy/osu-web#5061]
Self-explanatory, users can no longer delete or edit their modding posts. This is an essential part of the modding process. People sometimes make mistakes, those mistakes should be able to be deleted. Editing posts to add more information is another essential part of modding.

**Comment by cl8n** [comment]
if all's working well you can still edit/delete as long as it isn't resolved. see #4557

**Comment by nl-tatatat** [comment]
so... if you want to delete it, you have to reopen the issue? okay.

**Comment by cl8n** [comment]
> and now I'm stuck with comments reopening the discussion I can't delete either.

@notbakaneko I think it's because this query includes unresolves, when it should only be resolves. currently all of the posts between resolve~unresolve are locked unintentionally
https://github.com/ppy/osu-web/blob/6e9c54637e8b0c92838a23ffe3b0a717ea8d9fef/app/Models/BeatmapDiscussionPost.php#L180-L186

**Comment by nl-tatatat** [comment]
I don't think this is a useful feature. Why can't we at least append text to already resolved things? Sometimes there is something else you want to add to a post, but don't want it buried in a long discussion thread.
